thorlcr/activities/rollup/throllupslave.cpp

```diff
@@ -601,26 +613,35 @@ class CRollupGroupSlaveActivity : public CSlaveActivity, public CThorDataLink
         if (eoi)
             return NULL;
 
-        loop
+        try
         {
-            CThorExpandingRowArray rows(*this, queryRowInterfaces(input));
-            Owned<IRowStream> rowStream = groupLoader->loadGroup(input, abortSoon, &rows);
-            unsigned count = rows.ordinality();
-            if (0 == count)
+            loop
             {
-                eoi = true;
-                return NULL;
-            }
+                Owned<IRowStream> rowStream = groupLoader->loadGroup(input, abortSoon, &rows);
```

Comment:
Not for this PR, but returning a stream is a bit of a waste in this context.  There is also an issue with repeatedly registering/unregistering the callback.